src/components/pages/admin/fans/Index.js

```diff
@@ -208,6 +226,15 @@ class FanList extends Component {
 			});
 	}
 
+	onSearch(query) {
+		this.query = query;
```

Comment:
It might be worth putting the debounce code in this function, which will avoid the recursive lookup.
```suggestion
		clearTimeout(this.timeout);
		//Reset the page to 0 any time you are searching
		this.timeout = setTimeout(() => {this.refreshFans(query, 0)}, 500);
```